app/uk/gov/hmrc/helptosavefrontend/enrolment/EnrolmentStore.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.enrolment
+
+import cats.data.EitherT
+import com.google.inject.{ImplementedBy, Inject}
+import play.api.libs.json._
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.{BSONDocument, BSONObjectID}
+import uk.gov.hmrc.helptosavefrontend.enrolment.EnrolmentStore.{Enrolled, NotEnrolled, Status}
+import uk.gov.hmrc.helptosavefrontend.enrolment.MongoEnrolmentStore.EnrolmentData
+import uk.gov.hmrc.helptosavefrontend.util.NINO
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[MongoEnrolmentStore])
+trait EnrolmentStore {
+  import EnrolmentStore._
+
+  def get(nino: NINO): EitherT[Future,String,Status]
+
+  def put(nino: NINO, itmpHtSFlagSet: Boolean): EitherT[Future,String,Unit]
+
+}
+
+object EnrolmentStore {
+
+  sealed trait Status {
+    def fold[T](ifNotEnrolled: ⇒ T, ifEnrolled: Boolean ⇒ T): T = this match {
+      case e: Enrolled  ⇒ ifEnrolled(e.itmpHtSFlagSet)
+      case NotEnrolled  ⇒ ifNotEnrolled
+    }
+  }
+
+  case class Enrolled(itmpHtSFlagSet: Boolean) extends Status
+
+  case object NotEnrolled extends Status
+
+}
+
+class MongoEnrolmentStore @Inject()(mongo: ReactiveMongoComponent)(implicit ec: ExecutionContext)
+  extends ReactiveRepository[EnrolmentData, BSONObjectID] (
+    collectionName = "enrolments",
+    mongo = mongo.mongoConnector.db,
+    EnrolmentData.ninoFormat,
+    ReactiveMongoFormats.objectIdFormats)
+    with EnrolmentStore {
+
+  override def indexes: Seq[Index] = Seq(
+    Index(
+      key = Seq("nino" → IndexType.Ascending),
+      name = Some("ninoIndex")
+    )
+  )
+
+  def update(data: EnrolmentData)(implicit ec: ExecutionContext): Future[Option[EnrolmentData]] =
+    collection.findAndUpdate(
+      BSONDocument("nino" -> data.nino),
+      BSONDocument("$set" -> BSONDocument("imtpHtSFlagSet" -> data.imtpHtSFlagSet)),
```

Comment:
I named it that since the boolean was answering the question "has the ITMP flag been set for the user". But just `itmpFlag` works OK as well actually so I've changed that now. Also, there were typos where I wrote `imtp` instead of `itmp` - I've fixed those as well now